src/parse.rkt

```diff
@@ -72,6 +72,16 @@
     list-syntax
     list-syntax))
 
+(define/contract (elm-record-stx list-syntax list)
+                 (-> any/c 
+                     (flat-named-contract 'even-sized-list (compose even? length)) 
+                     syntax?)
+  (datum->syntax
+    list-syntax
+    #`(elm-record #,@(sequence->list (in-slice 2 list)))
+    list-syntax
+    list-syntax))
+
 ; {abc def ...} -> (elm-record (abc def) ...)
 (define (read-elm-record ch in src ln col pos)
   (define list-syntax
```

Comment:
Yeah I'm also more used to `(let)`, the inner `(define)`s are equivalent IIRC (generate local bindings if inside `(define)` themselves, but they seem ... more imperative :)